src/babashka/neil.clj

```diff
@@ -357,36 +364,190 @@
+
+   [#{:latest-sha} #{:git/url :latest-sha}]
+   (fn [lib-sym lib-opts]
+     (let [url (or (:git/url lib-opts) (github-repo-http-url lib-sym))
+           sha (latest-github-sha (git-url->lib-sym url))]
+       {lib-sym {:git/url url :git/sha sha}}))
+
+   [#{:git/url :git/tag :git/sha}]
+   (fn [lib-sym lib-opts]
+     {lib-sym (select-keys lib-opts [:git/url :git/tag :git/sha])})})
+
+(def valid-lib-opts
+  "The set of all valid combinations of deps-new template deps opts."
+  (into #{} cat (keys lib-opts->template-deps-fn)))
+
+(defn- deps-new-cli-opts->lib-opts
+  "Returns parsed deps-new template deps opts from raw CLI opts."
+  [cli-opts]
+  (-> cli-opts
+      (set/rename-keys {:sha :git/sha})
+      (select-keys (into #{} cat valid-lib-opts))))
+
+(defn- invalid-lib-opts-error [provided-lib-opts]
+  (ex-info (str "Provided invalid combination of CLI options for deps-new "
+                "template deps.")
+           {:provided-opts (set (keys provided-lib-opts))
+            :valid-combinations valid-lib-opts}))
+
+(defn- find-template-deps-fn
+  "Returns a template-deps-fn given lib-opts parsed from raw CLI opts."
+  [lib-opts]
+  (some (fn [[k v]] (and (contains? (set k) (set (keys lib-opts))) v))
+        lib-opts->template-deps-fn))
+
+(defn- template-deps
+  "Returns a tools.deps lib map for the given CLI opts."
+  [template cli-opts]
+  (let [lib-opts (deps-new-cli-opts->lib-opts cli-opts)
+        lib-sym (edn/read-string template)
+        template-deps-fn (find-template-deps-fn lib-opts)]
+    (if-not template-deps-fn
+      (throw (invalid-lib-opts-error lib-opts))
+      (template-deps-fn lib-sym lib-opts))))
+
+(def create-opts-deny-list
+  [:deps-file :dry-run :git/sha :git/url :latest-sha :local/root :sha])
+
+(defn- deps-new-plan
+  "Returns a plan for calling org.corfield.new/create.
+
+  :template-deps - These deps will be added with babashka.deps/add-deps before
+                   calling the create function.
+
+  :create-opts   - This map contains the options that will be passed to the
+                   create function."
+  [cli-opts]
+  (let [create-opts (merge {:template "scratch"}
+                           (apply dissoc cli-opts create-opts-deny-list))
+        tpl-deps (when (and bb? (not (built-in-template? (:template create-opts))))
+                   (template-deps (:template create-opts) cli-opts))]
+    (merge (when tpl-deps {:template-deps tpl-deps})
+           {:create-opts create-opts})))
+
+(defn- deps-new-create [create-opts]
+  ((requiring-resolve 'org.corfield.new/create) create-opts))
+
+(defn print-new-help []
+  (println (str/trim "
+Usage: neil new [template] [name] [target-dir] [options]
+
+Runs the org.corfield.new/create function from deps-new.
+
+All of the deps-new options can be provided as CLI options:
+
+https://github.com/seancorfield/deps-new/blob/develop/doc/options.md
+
+Both built-in and remote templates are supported. Built-in templates use
```

Comment:
Maybe "remote" is not the right word here since `:local/root` can be used?

Reply:
Maybe `external` is a better word?

Reply:
Yep, that sounds better to me!